### PR TITLE
Migrate to GitHub Actions

### PR DESCRIPTION
**Pull request by m-kn**
### Background
Travis costs money now or we have to regularly (monthly?) beg for OSS allowance or something

### Changes
What this PR does in Detail
  - Add rudimentary GitHub action "workflows" to
    - build on push to any branch
    - deploy on push of a SemVer tag (including snapshots)
  - Remove files related to travis deployment
  - Update the doc/README.md to reflect the deployment changes
  - Clean up the pom.xml
    - Update versions of dependencies & plugins
    - Remove unused dependencies
    - No longer inherit from com.mercateo.oss.parent:oss-parent-pom (temporarily)
        - our parent hasn't been updated for 4 years
        - it still depends on org.sonatype:oss-parents
        - which hasn't been updated in ~10 years, is deprecated and archived

### CODE REVIEW

**Comment by uweschaefer** [comment]
some inpiration: there is a DSL for GH Actions that makes using them much easier:
https://krzema12.github.io/github-workflows-kt/user-guide/getting_started/

example: 
https://github.com/factcast/factcast/blob/master/.github/kts/

**Comment by m-kn** [comment]
@uweschaefer

> some inspiration: there is a DSL for GH Actions that makes using them much easier: https://krzema12.github.io/github-workflows-kt/user-guide/getting_started/
> 
> example: https://github.com/factcast/factcast/blob/master/.github/kts/

Thx for tip! Problem is, I don't know an ounce of Kotlin :sweat_smile: and just wanted to be able to build & potentially release this project again, even if it's a q&d hack. Maybe a more permanent solution for the entire Org will be based on that DSL.

**Comment by uweschaefer** [comment]
@m-kn i understand you want to limit the scope of the change. however - for the record - you don't need any kotlin knowledge to use and profit from the DSL.

**Review by zohaibjawaid — COMMENTED** [review]
Overall looks good.

File: doc/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Release HowTo
+1. Merge your changes into the master branch
+1. Execute `mvn release:prepare`
+    - prompts you for a tag, choose a SemVer conform tag that reflects the changes you made
+    - creates a new commit with the new tag
+    - tags the new commit
+    - creates another ccomit
```
[the remark this answers is not in the document]
Comment (m-kn) [reply doc/README.md]: fixed

**Review by zohaibjawaid — APPROVED** [review]

**Pull request merged**